src/fusion_power_plant.cc

```diff
@@ -10,14 +10,97 @@ std::string FusionPowerPlant::str() {
   return Facility::str();
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void FusionPowerPlant::EnterNotify() {
+  //Pseudocode implementation of EnterNotify()
+  cyclus::Facility::EnterNotify();
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  //Truncated for readability. It goes on Tritium Storage and gets started, though.
+  fuel_startup_policy.Init(&tritium_storage).Set(fuel_incommod).Start();
+  
+  //Truncated for readability
+  blanket_fill_policy.Init(&blanket).Set(blanket_incommod).Start();
+
+  //Tritium Buy Policy Section:
+  if (refuel_mode == "schedule") {
+    IntDistribution::Ptr active_dist =
+        FixedIntDist::Ptr(new FixedIntDist(1));
+    IntDistribution::Ptr dormant_dist =
+        FixedIntDist::Ptr(new FixedIntDist(buy_frequency - 1));
+    DoubleDistribution::Ptr size_dist =
+        FixedDoubleDist::Ptr(new FixedDoubleDist(1));
+    //Do not start the policy yet.
+    fuel_refill_policy
+        .Init(&tritium_storage, &fuel_tracker,
+              buy_quantity, active_dist, dormant_dist, size_dis)
+        .Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    //otherwise we just do it normally, but still don't start it.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else {
+    //This can be more sophisticated later.
+    throw cyclus::KeyError("Refill mode not recognized!");
+  }
+
+  tritium_sell_policy.Init(&tritium_excess).Set(fuel_incommod).Start();
+  helium_sell_policy.Init(&helium_storage).Set(he3).Start();
+
+}
+
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tick() {
-  std::cout<<"Hello"<<std::endl;
+  //pseudocode implementation of Tick():
+  
+  if (CheckOperatingConditions()) {
+    fuel_startup_policy.Stop();
+    fuel_refill_policy.Start();
+    SequesterTritium();
+    OperateReactor();
+  } else {
+    //Some way of leaving a record of what is going wrong is helpful info I think
+    Record(Error);
+  }
+
+ 
+  DecayInventories();
+  ExtractHelium();
+  MoveExcessTritiumToSellBuffer();
+
+  //Redone blanket implementation:
+  if (BlanketCycleTime()) {
+    if (blanket->quantity() >= blanket_turnover) {
+      blanket_excess.Push(blanket->ExtractQty(blanket_turnover));
+      
+      //Temporary comment:
+      //Refill the in-core blanket. Blanket storage might be empty, though,
+      //which I think breaks the sim, so I've incuded try/catch to stop that
+      try {
+        blanket->Absorb(blanket_storage.Pop(blanket_turnover));
+      } catch (const std::exception& e) {
+        Record("Blanket Refill Error", e.what());
+        LOG(cyclus::LEV_INFO2, "Reactor") << e.what();
+      }
```

Comment:
Instead of failing this like we should either refill with what is available or not refill at all, and then the blanket quantity can be part of the condition for operation or not.

If we run out of blanket, we just don't operate.

Reply:
Makes sense to me